Merge MCMC parameter estimation toolbox

murrayrm opened the pull request:
Vipul has now finished his PhD and so I'm assuming we can merge this code into the master repository on BuildACell.

@vipulsinghal02 Is this ready to go?  If not, can you post here what needs to be added/fixed?

Commit:
first draft of the MCMC toolbox

Commit:
temp commit before I delete the sysID stuff

Commit:
deleted sys ID

Commit:
changed the ATP deg parameter setup, and named the NTP consumption binding second step parameters differently.

Commit:
added the globalize params function

Commit:
modified the txtl_init files

Commit:
fixed globalize params

Commit:
parameter globalization and rule creation in txtl_transcription and txtl_translation

Commit:
added the mcmc config file

Commit:
an mcmc toolbox for simbiology and in particular the txtl modeling toolbox

Commit:
added a gitignore file for the projects directory

Commit:
changed the readme file

Commit:
changed sbiointegrable to only tolerate 50% non integrability - down from 75%

Commit:
Massive developments to the MCMC toolbox

Commit:
added a file on VNPRL2011, and lots of stubs for the calibration correction work

Commit:
getting ready to run vnprl on nessa and lambda

Commit:
added a bunch of project and mcmc_info files for artificial data MCMC

Commit:
updates constgfp3ii mcmc info file for running on linux, 200 walkers, 40 iterations, parallel, thinning 10, step size 1.3

Commit:
updates constgfp3ii apr 1 7 pm PST for running on nessa and lambda

Commit:
updates constgfp3ii mcmc_info smaller step size and number of simulation points, apr 2, 3 pm

Commit:
turned proj_protein_constgfp3ii_linux into a function so one does not need to do a commit every time.

Commit:
optional ability to add a previous run as a starting point

Commit:
changes to linux file

Commit:
changes to file again

Commit:
added standard deviation tot he linux file

Commit:
bugfix in transcription and translation file, some setup of the covariation stuff

Commit:
an attempt at the new AGTP regenration system

Commit:
ommit log

- added the line:
ATP_regeneration_rate, Numeric,0.0002, !TODO: document,
to E30VNPRL_config and Emcmc2017_config

and
        ATP_regeneration_rate;
to txtl_reaction_config.m

- also made some files:
proj_acs_dsg2014_regen_A.m
mcmc_info_dsg2014_regen_A.m
data_dsg2014_full.m

Commit:
fixed a bug in the mcmc_info file. zzz

Commit:
first set of nessa and lambda runs done. Now extended the parameter ranges of the tx elongation, agtp regeneration system turnoff time, AA_F, UTRKd parameters. Going to initialize with the previous distributions and rerun at a much larger step size, and a much higher standard deviation (so that there is more mixing. only doing 20 iterations. going to analyze results. then rerun on new step sizes and standard deviations. eventually want to have a small SD and a small SS. This is the cooling effect. Simulated annealing. it should not violate detailed balance since eventually we will still to one set of conditioned, and this initial phase can just be thought of as a randomization phase where we try to find some initial distribution of points. This randomization is not totally random, indeed we are messily searching a distribution of parameters that fit the data, albeit poorly. "
git commit -m "first set of nessa and lambda runs done. Now extended the parameter ranges of the tx elongation, agtp regeneration system turnoff time, AA_F, UTRKd parameters. Going to initialize with the previous distributions and rerun at a much larger step size, and a much higher standard deviation. so that there is more mixing. only doing 20 iterations. going to analyze results. then rerun on new step sizes and standard deviations. eventually want to have a small SD and a small SS. This is the cooling effect. Simulated annealing. it should not violate detailed balance since eventually we will still to one set of conditioned, and this initial phase can just be thought of as a randomization phase where we try to find some initial distribution of points. This randomization is not totally random, indeed we are messily searching a distribution of parameters that fit the data, albeit poorly. "

"

Commit:
made come changed to the explore regen A file

Commit:
new gitignore

Commit:
gitignore fix

murrayrm commented:
There's a bunch of junk in this repository (crash files, etc) that should be removed before we merge.  

@vipulsinghal02 Can you fix these when you have a chance?

murrayrm commented on Crash_logs_14_10_17_22_01_18/matlab_crash_dump.17071-1:
```diff
@@ -0,0 +1,117 @@
+
```
Crash logs don't belong in the repository.

vipulsinghal02 commented:
Ok to the merging into the main branch. Will remove the crash logs and other junk and add the relevant regular expressions to the gitignore file. Should probably also include all the png files  / any data files etc. 

On the broader question of what needs to be tied up before the merge can be done, I think I have to go through the code base and separate out the core toolbox + a set of minimal and well documented examples, and just merge that, and have all the examples and use cases still under development as a separate branch (there are a bunch of examples I was working on to try out calibration-corrections at various parameter sharing configurations / levels of modeling.. all partially done so not worth including in the main branch.)

Timeline wise: have a few interviews with labs at ASTAR over the next week or so, but after that I can take some time to look at sorting through this, writing out a documentation file / user guide etc. Should take a ~4 weeks working on this part time.

murrayrm commented:
Sounds like a plan.  I'd like to get something minimal merged quickly, so that others can access it.  Is there a way to create a branch on your end with just the set of things that are already working and then we can do a pull request off of that.

Commit:
commit of untracked files, getting ready for setting up a minimal working code branch, which we can then merge into BuildACell

vipulsinghal02 commented:
Looks like we will have to be a bit careful here, since along with the mcmc_simbio code, there are also a bunch of changes in the main toolbox (a lot of code cleanup, added the ATP regeneration mechanism, etc). I would suggest making a new branch in the BuildACell/txtlsim repo and calling it something like "from_vipul" or "mcmc_simbio_and_other_changes". Then I can do a pull request from my repo to that branch, and we can look at the changes in that branch, and only merge what we want / see if everything is OK to merge. What do you think?  

More generally, I think we should discuss how to our collaborative workflow should be structured. I like the fork-pull request methodology, with branches for every feature in my fork. However, there are a couple of things to clarify if we try this: 

1. If, over a few months, I open a bunch of branches, make the relevant changes to each branch, and remerge into my fork's master branch, my version of the toolbox will have advanced quite a bit. So now after a few months, if we try to do a pull request, there is presumably a lot of discussion to be had: one discussion for each branch / feature that was added. This is the situation we are in right now. 
2. An alternative to this is that after every branch that gets completed on my end, and merged into my master, I do a pull request to the BuildACell/txtlsim repo. Can do this, but then we might decide that some feature that I add into my fork is not really needed or wanted in the BuildACell/txtlsim repo. So now if I merge my branch in, from that point on, my repo can no longer be pulled into the BuildACell repo, right? I am guessing that for the foreseeable future, a situation where I make a feature that does not get incorporated into the BuildACell version is probably unlikely, so this is not really a problem. Actually, with the MCMC toolbox, a version of this situation has already arisen: we want to merge the mcmc_toolbox, but not necessarily any of the changes I made to the rest of txtlsim toolbox.

vipulsinghal02 commented:
Another thing we can try is: I check out the BuildACell/txtlsim repo again, and add (a minimal version of) the mcmc_simbio toolbox to this new working tree, and we just do a pull request off that. That should be clean, but I think that means that we can no longer pull from my current fork, since the mcmc_toolbox files might be in conflict... Not sure, we could try, and this could be a learning experience for me. :-)

If we do try this, then I would just treat this new forked repo of mine as my main repo, and carefully open individual branches for each small change, and merge them one my one, and all the while we could be pulling each new merge into BuildACell, with discussion / explanation to document everything... Could be clean and nice, and can be done at a slower pace, since the mcmc_simbio is already in the BuildACell at this point, but it will take more effort (i think..)

vipulsinghal02 commented:
Another update (sorry for the spam!). Here is a list of the files that got modified/added/deleted in my fork wrt the BuildACell/txtlsim repo that are *not* part of the mcmc_simbio toolbox (and would therefore need discussion if they change anything significant (and a few of them do.. like txtl_transcription, txtl_translation, the ATP regeneration etc)). It is really not as bad as I had feared, so I think this should be doable quickly. (Tue onwards I have some time to do this)

Will add the Crash logs, png files, html files etc to the .gitignore and delete / untrack, and apart from the txtl_transcription, txtl_translation, the ATP regen, new parameters in the config files, and a couple of other minor things (all of which I can say a few words about), there aren't any other changes that can affect the core functioning of the BuildACell/txtlsim repo. 

M       .gitignore
A       Crash_logs_14_10_17_22_01_18/matlab_crash_dump.17071-1
A       Crash_logs_14_10_17_22_01_18/matlab_crash_dump.2495-1
A       Crash_logs_14_10_17_22_01_18/matlab_crash_dump.38232-1
A       Crash_logs_14_10_17_22_01_18/matlab_crash_dump.6005-1
A       Crash_logs_14_10_17_22_01_18/matlab_crash_dump.628-1
A       auxiliary/globalize_params.m
A       auxiliary/listmodelparts.m
M       auxiliary/plotCustomSpecies2.m
M       components/txtl_prom_p70.m
M       config/E30VNPRL_config.csv
M       config/E30_config.csv
M       config/E32_config.csv
M       config/E9_config.csv
A       config/Emcmc2017_config.csv
A       config/Emcmc2018_config.csv
M       core/txtl_addreaction.m
M       core/txtl_enzyme_resource_degradation.m
M       core/txtl_mrna_degradation.m
D       core/txtl_plot_gui.fig
M       core/txtl_reaction_config.m
M       core/txtl_runsim.m
M       core/txtl_transcription.m
M       core/txtl_translation.m
A       cov_extract2.jpg
D       doc/extractTODOlist
D       doc/txtl_template.m
M       examples/geneexpr.m
A       examples/geneexpr_test_regen_mode.m
M       html/txtl_tutorial.html
D       html/txtl_tutorial.png
D       html/txtl_tutorial_01.png
D       html/txtl_tutorial_02.png
D       modules/paramest/paramestCostFcn.m
D       modules/paramest/parameter_estimation_patternsearch.m
D       sysID/MCMC_example.m
D       tests/RNAdegradation_test.m
D       tests/atp_test.m
D       tests/geneexpr_DNAsweep.m
A       txtl_de_init.m
M       txtl_init.m
M       txtl_tutorial.m
A       txtl_tutorial_1.m

Commit:
deleted a lot of the files that were a work in progress, so that we can make sure the files that remain are fully functional and well documented examples. This will be the minimal toolbox for the mcmc_simbio toolbox that we can push to the BuildACell repository.

Commit:
deleted a lot of the files that were a work in progress, so that we can make sure the files that remain are fully functional and well documented examples. This will be the minimal toolbox for the mcmc_simbio toolbox that we can push to the BuildACell repository.

Commit:
Merge branch 'minimal_mcmc_simbio' of https://github.com/vipulsinghal02/txtlsim into minimal_mcmc_simbio

I dont know what is happening. Just trying to get my local copy to be in line with the branch in the remote. We will see what happens.

Commit:
Added a readme file in markdown format. Not complete yet; Still need to make the mcmc_tutorial and update the  TXTL_tutorial file, and update with a link to the thesis for further documentation

Commit:
Added a readme file in markdown format. Not complete yet; Still need to make the mcmc_tutorial and update the  TXTL_tutorial file, and update with a link to the thesis for further documentation

Commit:
try again with the readme file
I am trying a git commit --amend to see if I can get it to work without a problem. Had conflicts in the past. Not sure what is going on.

Commit:
tutorial files for mcmc_simbio in MATLAB markup

Commit:
A lot of write up to the mcmc_info.m helpfile. Will point to this file from all other files whenever something is to be explained about the mcmc_info struct. I imagine I will be updating this file with more info as I write the demo examples.

Commit:
added tutorials I-III, an mcmc_info help file, a data_info help file, and the mcmc_info constructor files for the three tutorials

Commit:
updated the readme file, and deleted some image files

Commit:
added support functions needed for this toolbox to function (there were in a common directory on my computer)

vipulsinghal02 commented:
@murrayrm 

Just created a new pull request from my minimal_mcmc_simbio branch to a branch of the same name in the BuildACell\txtlsim repo. I think we should do this first, and then only merge into master once we are sure everything is fine. For now people can simply checkout the minimal_mcmc_simbio branch to explore. 

Vipul #43

Commit:
changes to the source code to add weighting to trajectories, and the tierra data fitting files from Aug 2018. Here we tried to fit the data Abel sent us.

Commit:
modified gitignore to include data tiles, png, tiff, graphclick files

Commit:
testing on aws, first commit

Commit:
some changes to source files. refacter i next commit. this is from the aws instance to get things to work. also, need to check if the gen residuals is working properly.

Commit:
modifications to proj_acs_dsg2014_regen_A.m

Commit:
changed non integrability limit to 80%

Commit:
changed back to 67%

Commit:
sorry this is the one

Commit:
added display msg to integrableLHS.m

Commit:
more displays

Commit:
moving to debugging locally, AWS capability is present now. We can do both spot instances with hibernation, and on demand pricing. Next step is to get some project files set up, and have everything bug free and tested locally or on small aws instances. Then we can run things on large c5.8xlarge type instances either spot or on demand.

Commit:
changes to tutorialII, and ran it on the cluster

Commit:
not sure what happened here... why was a conflict generated in the gitignore? more importantly, why did the files VNPRL etc show up in the minimal mcmc simbio branch? very confusing, will investigate

Commit:
really screwing up the pushing here. I think I am mixing up the pulling and pushing between the master and minimal_mcmc_simbio branches... sigh. need to go over things later and fix manually.

Commit:
added the ability to start from a previous run's endpoint.

Commit:
bugfix

Commit:
bugfix1

Commit:
fixed gen residuals

Commit:
tutorial II made to start from previous run

vipulsinghal02 committed:
starting to fix some of the damage dome over the last couple of days... and then to retry the things I was trying to do (namely, to get the initial walker computationns parallelized in gwmcmc

Commit:
The error was that with the removal of the step which makes logical functions into non logical functions that turn false into -Inf, suddenly 0 became a valid point, leading to acceptance of points outside the parameter ranges. I have fixed this, and also added an experimental version of parallel computing of the initial walker positions.

vipulsinghal02 committed:
added parfor in the initial walker positions in gwmcmc_vse

vipulsinghal02 committed:
some changes to the tutorial II file to get it ready for continuation on a large 48 or 96 core machine

Commit:
parallel computing benchmark

Commit:
updated the tutorial III file

Commit:
changed the number of workers to 60. so 12 extra workers implemented via hyperthreading, over the 48 physical cores.

Commit:
changed the number of workers to 46. this should keep cpu load low.

vipulsinghal02 committed:
Merged the the minimal_mcmc_simbio_v20190131 with minimal_mcmc_simbio (as two separate branches, just to be safe. Might be able to delete the original two branches once I verify everything is in working order.

Merge branch 'minmcmcv4' into minmcmcv3

# Conflicts:
#	mcmc_simbio/models_and_supporting_files/mcmc_info_constgfp3ii.m
#	mcmc_simbio/models_and_supporting_files/mcmc_info_constgfp3tetR1.m
#	mcmc_simbio/projects/proj_mcmc_tutorial_II.m
#	mcmc_simbio/projects/proj_mcmc_tutorial_III.m
#	mcmc_simbio/src/gwmcmc/gwmcmc_vse.m

vipulsinghal02 committed:
marged the master branch into the minimal mcmc simbio branch and how have a fully updated branch that I can work on and clean up / do simulations of for the acs_dsg_regenA/B series, the VNPRL series, and any of the other series.

vipulsinghal02 committed:
added the dsg files from the master branch, will now work on editing them and cleaning them up.

vipulsinghal02 committed:
added temperature ladder capability to the proj_acs_dsg2014_regen_A.m file. added initial draft of the corresponding plotting file.
added parallel computing capability to initial walker position computation.

vipulsinghal02 committed:
fixed a bug that was causing a file not found error. getting ready to run this on a 48 core machine on the Amazon EC2 cloud.

vipulsinghal02 committed:
added temperature and step size ladder capabilities for simulated annealing like approaches.

massive modifications to proj_acs_dsg_regen_A.m. Use this as a template for other files.

modified log file generation

vipulsinghal02 committed:
Created proj_acs_dsg2014_regen_C.m, which is a project that has all the forward rate parameter values fixed (as opposed to them being estimatable, as in regen_A), in the hope that this makes the parameter estimation problem more tractable.

vipulsinghal02 committed:
Not sure what good values are, just made the F rates whole numbers between 1 and 10. Can try a few iterations and see what works... Can also try to hand fit.

vipulsinghal02 committed:
Fixed a bug in mcmc_get_walkers that was not gerring the walkers for more than the first timestamp when multiple timestamps were specified.

made the project file such that a numeric id of the current temperature is added to filenemes.

Still working on the explore and analysis scripts.

Commit:
added the ability to specify a particular iteratino ID from a previous simulation.

vipulsinghal02 committed:
temperature ladder defaults

vipulsinghal02 committed:
Merge branch 'minmcmcv5a' of https://github.com/vipulsinghal02/txtlsim into minmcmcv5a

vipulsinghal02 committed:
fixed a bug where there was initial DNA concentration. This was causingn the purified RNA experiment to fail.

vipulsinghal02 committed:
added an option to specify the step size multiplier for each iteration.

vipulsinghal02 committed:
added a project file regen_A1, which has expanded ranges of parameters, from looking at the posterior density plots.

vipulsinghal02 committed:
added the C1 project:
Used the sims
%         ts2temp = '20190205_145620_2_218';
%         ts3temp = '20190205_222248_1_218';
%         ts4temp = '20190206_043320_1_218';
%         ts6temp = '20190207_154246_1_218';
to look at the posterior distributions and fixed 12 ot of the 23 params.

Anything that was a forward rate set to a value determined by the posterior distribution: usually something close to 1 worked.

A couple of parameters that are sharply peaked, like RNA deg catalysis rate, and the time of ntp degratation turn on, we also set, since every dimension we fix makes the mcmc exponentially easier (curse of dimensionality)

fixed 12 (!) parameters:
activeNames2(fixedParamsIX,[1:2])

  12×2 cell array
    {'AGTPdeg_time'              }    {[    14328]}
    {'AGTPdeg_rate'              }    {[   0.0002]}
    {'AGTPreg_ON'                }    {[     0.02]}
    {'TXTL_P70_RNAPbound_F'      }    {[   4.4817]}
    {'TXTL_NTP_RNAP_1_F'         }    {[        1]}
    {'TXTL_NTP_RNAP_2_F'         }    {[        1]}
    {'TXTL_RNAdeg_F'             }    {[        1]}
    {'TXTL_RNAdeg_kc'            }    {[0.0045166]}
    {'TXTL_PROT_deGFP_MATURATION'}    {[   0.0023]}
    {'TXTL_UTR_UTR1_F'           }    {[  0.81873]}
    {'TL_AA_F'                   }    {[  0.74082]}
    {'TL_AGTP_F'                 }    {[  0.30119]}

and so only estimating (14!) params, -> this should be a lot more manageable.
    {'TX_elong_glob'                  }
    {'TXTL_P70_RNAPbound_Kd'          }
    {'TXTL_RNAPBOUND_TERMINATION_RATE'}
    {'TXTL_NTP_RNAP_1_Kd'             }
    {'TXTL_NTP_RNAP_2_Kd'             }
    {'TXTL_RNAdeg_Kd'                 }
    {'RNAP'                           }
    {'RNase'                          }
    {'TL_elong_glob'                  }
    {'TXTL_UTR_UTR1_Kd'               }
    {'TL_AA_Kd'                       }
    {'TL_AGTP_Kd'                     }
    {'TXTL_RIBOBOUND_TERMINATION_RATE'}
    {'Ribo'                           }

vipulsinghal02 committed:
fixed a bug where the master vector was not being initialized properly. (it was set ot all 1's, did not matter too much for regen_A and regen_A1, since most of the species there were estimated, but in regen_C, having a lot of species being 1 was too much for the model to be able to fit the data.

vipulsinghal02 committed:
removed dose weighting, and also flipped the experiment weighting. I had is wrong before, where the experiment with less signal had a lowered weighting.

vipulsinghal02 committed:
minor bugfix

vipulsinghal02 committed:
made some changes to the mcmc_info files to reject the dose weighting.

vipulsinghal02 committed:
unfixed the deg turn on time parameter.

vipulsinghal02 committed:
proj_acs_dsg2014_regen_D: minimal 10 parameter version, with the rest of the parameters from a hand fit (inspired by runs already done) see analysis_regen_C.m for the vector mod1 for full details.

vipulsinghal02 committed:
project regen_G: a very very reduced set of just 6 parameters. the RNA deg parameters fixed (and the corresponding model info - data info removed. Also the termination rates fixed. So only the following 6 params are estimated:
%     {'TX_elong_glob'                  }
%     {'TXTL_P70_RNAPbound_Kd'          }
%     {'RNAP'                           }
%     {'TL_elong_glob'                  }
%     {'TXTL_UTR_UTR1_Kd'               }
%     {'Ribo'                           }

vipulsinghal02 committed:
added regen E and F, that are just variations involving 11 and 13 parameters:
regen_E is just the following 11:
%     {'TX_elong_glob'                  }
%     {'TXTL_P70_RNAPbound_Kd'          }
%     {'TXTL_RNAPBOUND_TERMINATION_RATE'}
%     {'TXTL_RNAdeg_Kd'                 }
%     {'TXTL_RNAdeg_kc'                 }
%     {'RNase'                          }
%     {'RNAP'                           }
%     {'TL_elong_glob'                  }
%     {'TXTL_UTR_UTR1_Kd'               }
%     {'TXTL_RIBOBOUND_TERMINATION_RATE'}
%     {'Ribo'                           }

regen_F is the following 13
%     {'TX_elong_glob'                  }
%     {'AGTPdeg_time'                   }
%     {'AGTPdeg_rate'                   }
%     {'TXTL_P70_RNAPbound_Kd'          }
%     {'TXTL_RNAPBOUND_TERMINATION_RATE'}
%     {'TXTL_RNAdeg_Kd'                 }
%     {'TXTL_RNAdeg_kc'                 }
%     {'RNAP'                           }
%     {'RNase'                          }
%     {'TL_elong_glob'                  }
%     {'TXTL_UTR_UTR1_Kd'               }
%     {'TXTL_RIBOBOUND_TERMINATION_RATE'}
%     {'Ribo'                           }

vipulsinghal02 committed:
upweighted the tx and tl experiment relative to the rna deg experiment. (the rna deg exp to tx+tl experiment used to be 1:0.03, now it is 1:0.3)

vipulsinghal02 committed:
two larger parameter estimation problems, H and I, too. 19 and 15 parameters respectively.

vipulsinghal02 committed:
F1 is F with expanded allowable parameter ranges.

vipulsinghal02 committed:
added a vnprl project. It is basically like the ACS 2014 paper, but with the protein expression and mrna levels scaled down to match what is claimed in Vincent Noireaux's PRL paper.

vipulsinghal02 committed:
added the test015 data and estimation files -- this is the investigation of calibration correction shown in the thesis. -- ptet GFP calibration experiments, and tetR repression test data.

also started adding the Zach IFFL data files, but not done with those yet.

vipulsinghal02 committed:
extended the parameter ranges

vipulsinghal02 committed:
the first correction step of the tetR data. (redoing the thesis stuff on real experimental data)

vipulsinghal02 committed:
dosedvals bugfix in mcmc_info

vipulsinghal02 committed:
another bugfix

vipulsinghal02 committed:
flipped the tetR and GFP dose. cannot use unflipped... not sure why. need to look at how I wrote the code.... (its been so long)

vipulsinghal02 committed:
extended the ranges a bit after looking at the plots

vipulsinghal02 committed:
fixed the forward rate parameters in the corr step 1.

vipulsinghal02 committed:
made some changes ot the params... opefully removes some of the ill conditioning. ideally start the LHS from a small cube around good values, then once the small scale structure is dileanated, can expand the limits, and explore a much larger space.

vipulsinghal02 committed:
using uM data with nM DNA conc was messing up the tetR repression, where the uM tetR protein had to bind the nM dna... always convert to consistent units!!! need to redo the calib step...

vipulsinghal02 committed:
changed the correction files with the new ESPs estimated using the corrections in the calibration scripts in the previous commits (basiclaly, the protiens are now in nM, same units as the DNA.)

vipulsinghal02 committed:
trying a more relaxed fixing scheme -- only the forward dimerization rate is fixed. MCMC should be able to handle this!

vipulsinghal02 committed:
make the ranges smaller again to avoid non integrability.

vipulsinghal02 committed:
did not update the est names.... fixed that bug

vipulsinghal02 committed:
fixed more parameters... the correction step was not really working

vipulsinghal02 committed:
fixing the kfdg in the calibration experiments, and going to try that.

vipulsinghal02 committed:
didnt set it up properly. corrected.

vipulsinghal02 committed:
working on figuring out a better scheme for generalization to the test circuit. see analysis corr1_Ffix2

vipulsinghal02 committed:
going to try doing the calibrations with both the extract datasets the same. bobby's idea.

vipulsinghal02 committed:
Bug in the proj file--- forgot to change the mcmc_info command in the proj file... zzz.

vipulsinghal02 committed:
Commit message: Large commit for phase 1 of the IFFL data fitting using Zach's data.

Changes:

* major bugfix: in exportmobj, the dose names in the model info struct now get ordered to match the model object. this was never a problem in the past because the number of dosed species was small.

* fixed a bug in the file txtl_dimerize that had the parameter names not follow the naming convention.

* fixed a bug in txtl_translation that had a 2 in front of the AGTP.

Added files
mcmc_info_ZSIFFL_mtet_phase1.m <--- done: estimate the inner params of the tetR system.
mcmc_info_ZSIFFL_mtet.m <---- not done: estimate the outer parameters of the tetR system
mcmc_info_ZSIFFL_training.m <---- not done
proj_ZSIFFL_mtet.m <--- done.
proj_ZSIFFL_training.m <---- not done,

vipulsinghal02 committed:
minor bugfix

vipulsinghal02 committed:
phase 1b: fixing more parameters, expanding the range for others. also started the stubs for phase 2 and the full IFFL training.

vipulsinghal02 committed:
further readjusted the parameter ranges.

vipulsinghal02 committed:
more param range relaxations

vipulsinghal02 committed:
further extended the range of parameter values, and fixed a bug in the data info.

vipulsinghal02 committed:
SET UP PHASE 2 WITH THE RIGHT PARAMETERS

vipulsinghal02 committed:
added trajectory plotting code, and fixed bug in phase 2 mcmc info file where the ranges were not right.

vipulsinghal02 committed:
extended the limits of RNAP and the tetR sequestration parameters.

vipulsinghal02 committed:
extended the ranges of a bunch of parameters: Pol, Ribo, Pol KD (tet), rnase, atp deg time.

vipulsinghal02 committed:
added the full IFFL training example. (proj and mcmc_info file)

vipulsinghal02 committed:
bugfix in the full iffl training file.

vipulsinghal02 committed:
the plas system was not fitting well enough. there was not enough deGFP protein produced. I relaxed the termination rate for both tx ans tl, and finally got the plots to look somewhat reasonable by hand. Now to make the MCMC find these and other parameters that fit.

vipulsinghal02 committed:
made the parameter ranges for the unactivated plas promoter Kd broader. This is to allow weaker association of the RNAP to the unactivated promoter.

vipulsinghal02 committed:
Added full training B files:
"ZSIFFL_training_fullB"

that have fewer parameters (lac and tet Kd no longer estimated) and tighter ranges.

vipulsinghal02 committed:
added the file training_fullC, which has expanded parameter ranges, and also estimates many more parameters.

estParamsIX = [1 2 7 11 16 21 23 28 31 33 35 37 39 40]';
estParams = activeNames(estParamsIX,1)

estParams =

  14×1 cell array

    {'TX_elong_glob'                  }
    {'TL_elong_glob'                  }
    {'TXTL_PTET_RNAPbound_Kd'         }
    {'TXTL_PTET_sequestration_Kd'     }
    {'TXTL_INDUCER_TETR_ATC_Kd'       }
    {'TXTL_PLAC_RNAPbound_Kd'         }
    {'TXTL_RNAPBOUND_TERMINATION_RATE'}
    {'TXTL_RIBOBOUND_TERMINATION_RATE'}
    {'RNAP'                           }
    {'Ribo'                           }
    {'TXTL_INDUCER_LASR_AHL_Kd'       }
    {'TXTL_PLAS_RNAPbound_Kd'         }
    {'TXTL_PLAS_TFBIND_Kd'            }
    {'TXTL_PLAS_TFRNAPbound_Kd'       }

instead of training_fullB's

estParamsIX = [23 28 31 33 35 37 39 40]';
estParams = activeNames(estParamsIX,1)

estParams =

  8×1 cell array

    {'TXTL_RNAPBOUND_TERMINATION_RATE'}
    {'TXTL_RIBOBOUND_TERMINATION_RATE'}
    {'RNAP'                           }
    {'Ribo'                           }
    {'TXTL_INDUCER_LASR_AHL_Kd'       }
    {'TXTL_PLAS_RNAPbound_Kd'         }
    {'TXTL_PLAS_TFBIND_Kd'            }
    {'TXTL_PLAS_TFRNAPbound_Kd'       }

vipulsinghal02 committed:
added training fullD: relative to training fullC, i fixed 3 parameters, after checking restrictions in analysis..training_fullC. The three parameters fixed were:
training_fullC_params = ...
11, 'TXTL_PTET_sequestration_Kd' , exp(-0.5)     ,[exp(-2)   exp(1)]
16	,'TXTL_INDUCER_TETR_ATC_Kd'  	, exp(-2)   ,[exp(-10) exp(5)]
35 , 'TXTL_INDUCER_LASR_AHL_Kd'           , exp(13), [exp(5) exp(15)]

vipulsinghal02 committed:
Added training E, that differs from training D in that the ktx and ktl are fixed from values found from training D. Note that in training D, we found the ktx and ktl to be exactly inversely proportional.

vipulsinghal02 committed:
massive changes to many files

vipulsinghal02 committed:
deleted large files

Commit:
the aws version where I run F3

vipulsinghal02 committed:
added IFFL training full F--> this has better parameter search ranges. has not been run yet. will run.

vipulsinghal02 committed:
renamed the unmodified vnprlF3 to keep it.

vipulsinghal02 committed:
Merge branch 'minmcmcv5a' of https://github.com/vipulsinghal02/txtlsim into minmcmcv5a

vipulsinghal02 committed:
bugfix in RNAP parameter ranges of vnprl F3

vipulsinghal02 committed:
ZSIFFL_E updated to F. This is basically using the distribution C parameter ranges, as described in the evernote note "035 APR 2020 VNPRL F2 (->F3)+ IFFL training E(->F)"

vipulsinghal02 committed:
added the ZSIFFL training F project file

vipulsinghal02 committed:
windiows compatible initialization.

vipulsinghal02 committed:
updated ZSIFFL training F

vipulsinghal02 committed:
Merge branch 'minmcmcv5a' of https://github.com/vipulsinghal02/txtlsim into minmcmcv5a

vipulsinghal02 committed:
all the project and mcmc info files from the runs done on the big workstation

vipulsinghal02 committed:
core characterization plots

vipulsinghal02 committed:
Merge branch 'minmcmcv5a' of https://github.com/vipulsinghal02/txtlsim into minmcmcv5a

# Conflicts:
#	generate_paper_figs/Core_characterization_plots.m

vipulsinghal02 committed:
restoring the runsummaries. a complicated thing between dropbox and github happened that caused it to be deleted: Take note.

my dropbox folder is available on both my mac (com1) and desktop (com2). This repo is available on dropbox, and therefore, on both com1 and com2. But really, this dropbox version should be thought of as the working copy for com1, NOT com2. We will call this working copy WC1.

Now, this repo also has a local non-dropbox working copy on com2. We will call this working copy WC2.

The problem occured when i edited a file (runsummareis.m) that was in WC1 on com2. but then from com2, i committed and pushed WC2 (which had the unedited file), and pulled this change onto com1's working copy, WC1. In this process, i overwrote all my changes to the file.

So really, word of advice:
1. whenever you make a change, puch it to the repo immediately, while it is still fresh.
2. Never change anything from com2 on the dropbox folder, and then expect that change to still be there when you push WC2 onto the remote repo.

Actually I think this problem occurred because I created the file on both WC1 and WC2, but did not commit either. I made changes to the one in WC1, but not WC2, and then committed the one in WC2, and then pulled this to overwrite the one in WC1, losing that copy in WC1.

vipulsinghal02 committed:
added the generate paper figs files. thse will be used to add the same files to tbe build a cell repo

vipulsinghal02 committed:
addressed merge conflict in the core charac file

vipulsinghal02 committed:
.

vipulsinghal02 committed:
Merge branch 'minmcmcv5a' of https://github.com/vipulsinghal02/txtlsim into minmcmcv5a

vipulsinghal02 committed:
gitignore and moved some things around

vipulsinghal02 committed:
incorporating changes to component files from the buildacell repo (which had the parameter values incorporated, and some small changes to a couple of component m files.

vipulsinghal02 committed:
tutorial and config files from the buildacell repo

vipulsinghal02 committed:
minor changes to geneexpr

vipulsinghal02 committed:
updated run summaries file

vipulsinghal02 committed:
First test of the file proj_vnprl_cluster to deploy the code on the cluster

vipulsinghal02 committed:
modifications to get it to run on the corona cluster. The current version throws an error where the rawdata_ACSDSG2014.m is not found.

vipulsinghal02 committed:
intermediate commit containing some changes to the tetR calibration correction files. More changes coming before this will be working. (this commit is to transfer the code between machines)

vipulsinghal02 committed:
mcmc runsim
